app/src/common/models/tag/tagModel.mock.ts

```diff
@@ -21,8 +21,8 @@ namespace common.models {
             return <Tag> new this().buildEntity(overrides, exists);
         }
 
-        public static collection(count:number = 10):Tag[] {
-            return <Tag[]>new this().buildCollection(count);
+        public static collection(count:number = 10, overrides:Object = {}, exists:boolean = true):Tag[] {
```

Comment:
Yea it would be desirable, however it seems that typescript doesn't allow for static method inheritance which means it has to be repeated for each extending class.

I think the only way to really handle the abstract would be to do some magic with a decorator, but then I don't think the typehinting would be able to work properly.